MovieViewer/MoviesViewController.swift

```diff
@@ -287,11 +301,126 @@ extension MoviesViewController: UITableViewDataSource {
         
         let cell = tableView.dequeueReusableCellWithIdentifier("MovieCell", forIndexPath: indexPath) as! MovieCell
         let movie = filteredMovies![indexPath.row]
+        
         cell.setData(movie)
         cell.setTheme()
         
+        // retrieve all favorites by id
+        self.ref!.child("favorite/\(movie.id!)").observeEventType(FIRDataEventType.Value, withBlock: { (snapshot) in
+            var favorite = snapshot.value!["isFavorite"] as! Bool
+            cell.backgroundColor = favorite ? UIColor.brownColor() : UIColor.clearColor()
+            let favoriteTitle = favorite ? "Unfavorite" : "Favorite"
+                
+            //configure left buttons as Favorite
+            cell.leftButtons = [MGSwipeButton(title: favoriteTitle, backgroundColor: UIColor.grayColor(), callback: {
+                (sender: MGSwipeTableCell!) -> Bool in
+                    
+                favorite = !favorite
+                self.ref?.child("favorite/\(movie.id!)/isFavorite").setValue(favorite)
```

Comment:
maybe this is a method that could be extacted to. `movie.toggleFavorite()` or something like that?